lib/Driver/ToolChains.cpp

```diff
@@ -1047,6 +1047,9 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
     break;
   case LinkKind::DynamicLibrary:
     Arguments.push_back("-shared");
+    if (getTriple().getArch() == llvm::Triple::ArchType::arm) {
+      Arguments.push_back("-Wl,-Bsymbolic");
```

Comment:
I'm hesitant to add a test for this as once 2.2 is released my plan is to transition to the gold linker for arm-linux, which not only will not require -Bsymbolic, it will expect other options.  Given that, as per my comment below re. early bail from linker.swift, these test aren't executed anyway, it makes sense to me to hold off until those other changes are made.